site_scons/prereq_tools/base.py

```diff
@@ -368,9 +368,10 @@ def get(self, subdir, **kw):
 *********************** ERROR ************************\n""" % comp)
             raise DownloadFailure(self.url, subdir)
 
-        commands = ['git clone %s %s' % (self.url, subdir)]
-        if not RUNNER.run_commands(commands):
-            raise DownloadFailure(self.url, subdir)
+        if not os.path.exists(subdir):
+            commands = ['git clone %s %s' % (self.url, subdir)]
+            if not RUNNER.run_commands(commands):
+                raise DownloadFailure(self.url, subdir)
```

Comment:
BTW, it's extremely slow to download patches as well, and in resolve_patches(), it always remove local patch file first then download patch file, I'm wondering if it could be improved to skip downloading if local patch file is existing?

Reply:
Yup, agreed, let me improve that as well!